### PR TITLE
Merge Authorization into main

### PR DESCRIPTION
**Pull request by Setanna**

### CODE REVIEW

**Comment by Setanna** [comment]
[Login](https://trello.com/c/vub0ZCPQ/5-login)

**Comment by Setanna** [comment]
[log out](https://trello.com/c/Q4wwu7C9/25-log-out)

File: lib/bin/api/api_logon.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+/// Check if the user is logged in
+Future<bool> isLoggedIn() async {
+  String? value = await storage.read(key: "token");
+  if (value != null) {
+    print("isLoggedIn Function: true");
+    return true;
+  }
+  print("isLoggedIn Function: false");
+  return false;
```
Comment (MathiasDuus) [inline lib/bin/api/api_logon.dart:38]: Wrap print in a check to see if debugmode or use debugPrint()

File: lib/login.dart [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+        'Accept': 'application/json'
+      },
+      body: jsonEncode(<String, String>{
+        'username': username,
+        'password': password,
+      }),
+    );
+  }
```
Comment (MathiasDuus) [inline lib/login.dart:298]: Maybe move it to the api_logon file :)

File: lib/widget/custom_drawer.dart [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    title: const Text("Sign Out"),
+                    onTap: () async {
+                      // Close Drawer
+                      Scaffold.of(context).closeDrawer();
+                      // Call api log out function
+                      await api.logOut();
+                    },
+                  ),
+                ]);
+              } else // Else display this listview
```
Comment (MathiasDuus) [inline lib/widget/custom_drawer.dart:65]: Maybe make it more descriptive, like what soes the else mean.

**Review by MathiasDuus — COMMENTED** [review]
Remeber to check formating of the code once in a while with Ctrl+Alt+L

File: lib/widget/custom_drawer.dart (hunk 3 above)
Comment (Setanna) [reply lib/widget/custom_drawer.dart:65]: Fair, will make it write something like "Else when not logged in, display this listview in drawer"

File: lib/bin/api/api_logon.dart (hunk 1 above)
Comment (Setanna) [reply lib/bin/api/api_logon.dart:38]: I will follow by example and use debugPrint().

File: lib/bin/api/api_logon.dart [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+/// Log user in
+Future<http.Response> login(final username, final password) {
+
+  var url = "";
+  // If debug mode is active, use the dev path.
+  if (kDebugMode) {
+    url = '${api.getBaseUrl()}/joen/api/auth/login';
+  } else {
+    url = '${api.getBaseUrl()}/api/auth/login';
+  }
```
Comment (MathiasDuus) [inline lib/bin/api/api_logon.dart:64]: Siden at produktion burde være up to date, lige nu så kan vi godt fjerne det her og bare bruge produktion.
Eller så skal checket måske rykkes til getBaseUrl() funktionen, dette er måske den bedre løsning.

File: lib/login.dart [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+import 'package:product_radar/sign_up.dart';
+
+class LoginPage extends StatefulWidget {
+  const LoginPage({super.key});
+
+  @override
+  State<LoginPage> createState() => _LoginPageState();
+}
+
+String password = "";
```
Comment (MathiasDuus) [inline lib/login.dart:15]: Den her værdi ser ikke ud til at blive brugt

Comment (Setanna) [reply lib/login.dart:15]: Tjækker og derefter fjerner hvis ikke brugt

File: lib/widget/custom_drawer.dart (hunk 3 above)
Comment (Setanna) [reply lib/widget/custom_drawer.dart]: it has been added

File: lib/login.dart (hunk 5 above)
Comment (Setanna) [reply lib/login.dart]: Den var unused og derfor fjernet

File: lib/bin/api/api_logon.dart (hunk 1 above)
Comment (Setanna) [reply lib/bin/api/api_logon.dart]: It has been changed to debugPrint().

**Review by MathiasDuus — APPROVED** [review]
Looks good to me

**Pull request merged**